packages/client/hmi-client/src/components/workflow/ops/simulate-ciemss/tera-simulate-node-ciemss.vue

```diff
@@ -160,14 +166,21 @@ Provide a summary in 100 words or less.
 	);
 	if (!datasetResult) {
 		logger.error('Error creating dataset from simulation result.');
+		state.errorMessage = {
+			name: 'Failed to create dataset',
+			value: '',
+			traceback: `Failed to create dataset from simulation result: ${runId}`
+		};
+		emit('update-state', state);
 		return;
 	}
 
+	// Note: this will also update state
```

Comment:
I do not understand your comment here, this will update the output-port's state. But you are **not** updating the node's state if that is the intent here. Can you elaborate/explain a bit?

Reply:
This comment is for the reader to understand that there is not a missing emit('update-state' ...)

I have updated to the form thanks to your explaination

```
	emit(
		'append-output',
		{
			output-state
		},
		state
```